src/outputs/outputs.cpp

```diff
@@ -252,20 +261,21 @@ Outputs::Outputs(Mesh *pm, ParameterInput *pin, SimTime *tm) {
         pnew_type = new FormattedTableOutput(op);
       } else if (op.file_type.compare("vtk") == 0) {
         pnew_type = new VTKOutput(op);
-      } else if (op.file_type.compare("rst") == 0) {
-        pnew_type = new RestartOutput(op);
-        num_rst_outputs++;
-      } else if (op.file_type.compare("ath5") == 0 || op.file_type.compare("hdf5") == 0) {
+      } else if (is_hdf5_output) {
+        bool restart = (op.file_type.compare("rst") == 0);
+        if (restart) {
+          num_rst_outputs++;
+        }
 #ifdef HDF5OUTPUT
-        pnew_type = new PHDF5Output(op);
+        pnew_type = new PHDF5Output(op, restart);
```

Comment:
Can we change this to a unique_ptr?

Reply:
Alas, no. This is a custom-made linked list and the pointers get passed around and manipulated a lot. At least it seems they all get properly freed in the destructor of the `Outputs` class. But I agree that this is not pretty. We should be using an STL container here. This should be relatively straightforward to fix, but it's beyond the scope of this PR. Might be a good "first issue".